test/test_dht.cpp

```diff
@@ -2059,5 +2121,103 @@ TORRENT_TEST(routing_table_extended)
 #endif
 }
 
+TORRENT_TEST(read_only_node)
```

Comment:
I think it's good to have an explicit negative test, and having it associated with the positive one seems reasonable. It seems like factoring it out would just be a matter of passing in the dht_settings object and the parsed array, and then assert outside of the factored out function.